alivee/place_search_engine/utils/place_search.py

```diff
@@ -0,0 +1,233 @@
+
+
+class PlaceSearch(object):
+	'''
+	Wrapped google place search service for our usage.
+	Documents: https://developers.google.com/places/web-service/intro
+
+	Usage example: Search all banks in 1000 meters radius area of location (37.7845782, -122.3910999)
+
+	placeSearchInstance = PlaceSearch(
+		location=(37.7845782, -122.3910999), 
+		key=key,
+	)
+
+	placeSearchInstance.nearbySearch(
+		place_type=PlaceServicePlaceType.BANK, 
+		radius=1000,
+	)
+	OR:
+	placeSearchInstance.radarSearch(
+		place_type=PlaceServicePlaceType.BANK, 
+		radius=1000,
+	)
+	'''
+	def __init__(self, location, key):
+		'''
+		Init method of place search class.
+
+		param location: A tuple (lat, lng) represents the location we want to set as the center of the search area.
+		param key: The Google API key, generated in google dev console, ask project dev for help.
+		'''
+		self.location = location
+		# NOTE: Need to handle exception in caller's method such as `ValueError: Invalid API key provided.`
+		self.client = googlemaps.Client(key)
+
+	def _getPlaceFilterKwargs(self, place_type, radius, keyword, name, rank_by, page_token):
+		'''
+		Private helper method to pre-check and create the filter kwargs in place search query, follow format of
+		https://github.com/googlemaps/google-maps-services-python/blob/master/googlemaps/places.py#L199
+
+		return: A map includes all params in place search query.
+
+		NOTE: Pre-check is needed to avoid obviously invalid API call, thus saving query count.
+		'''
+		filterKwargs = dict(location=self.location)
+
+		if place_type:
+			if not place_type in PLACE_SERVICE_SEARCH_PLACE_TYPES:
+				raise ValueError('Invalid place type, pre-check failed!'.format(place_type))
+			filterKwargs['type'] = place_type
+		if radius:
+			if radius > MAX_PLACE_SERACH_RADIUS_IN_METERS:
+				raise ValueError('Radius: {} exceeds maximum allowed value: {}, pre-check failed!'.format(
+					radius, MAX_PLACE_SERACH_RADIUS_IN_METERS))
+			filterKwargs['radius'] = radius
+		if keyword:
+			filterKwargs['keyword'] = keyword
+		if name:
+			filterKwargs['name'] = name
+		if rank_by:
+			if rank_by and radius:
+				raise ValueError('Cannot set both rank_by and radius, pre-check failed!')
+			filterKwargs['rank_by'] = rank_by
+		if page_token:
+			filterKwargs['page_token'] = page_token
+
+		return filterKwargs
+
+	def _checkReponseStatus(self, response):
+		'''
+		Private helper method to check if the status of google place search response is valid.
+
+		param response: Response of google place search query, notice `googlemaps` package already parses the 
+		JSON format response to dict - list format for us.
+		return: A none-boolean value represents whether the response is valid, `None` value means response is valid
+		but there is no real content in it.
+
+		NOTE: If response is not valid, exception will be raised instead of returning `False` value.
+		'''
+		if not response or not 'status' in response:
+			raise Exception('Invalid response!')
```

Comment:
Not sure how we wanna do exception handling. If we need to handle these specific exceptions, we may need consider making specific Exceptions other than using generic python exception.